week 6 router

gyeongwon1275 opened the pull request:
## 0301 

- [x] react-router 기본설정
- [x] header 
- [x] HomePage
- [x] About
- [x] NotFound

### 고민

- 식당 정보는 restaurantId 를 params 를 받은 후 mount 할 때 api 요청을 통해 받은 데이터를 출력하고자 함 
params 는 숫자만 받아야 할 것 같은데 문자등을 받았을 때 예외처리는 어떻게 할지 고민중 

- 링크로 url 이동하는 것과 사용자가 직접 url 입력해서 이동하는 것의 차이를 추후 확인하고자 함 

## 0302

- [x] restaurantDetail Page 기본화면
- [x] restaurantDetail 서버에서 데이터 받기 
- [x] webpack error 해결

### 고민 

현재 url 에서 params 를 제대로 받아오는지를 test 할 필요가 있을가? 

url 에서 params 를 받아오는 것은 react-router 의 책임이고 

현재 앱에선 받은 params 를 가지고 제대로 서버에 요청을 하는지만 확인하면 되지 않을까? 

생각한 테스트 방법은 

1> useParams 를 mock 처리 하기 

2> 공식문서대로 history 객체 이용하기 

이었으나 일단 공식문서대로 하기로 했다.  

공식문서( https://testing-library.com/docs/example-react-router/ )에서는 
아래함수처럼 history 객체를 Route component 에 전달하는 형태이다. 

```javascript
 function renderRestaurantDetail(path) {
    const history = createMemoryHistory();
    return render((
      <MemoryRouter initialEntries={[path]}>
        <Route
          exact
          history={history}
          path="/restaurants/:restaurantId"
          component={RestaurantDetailContainer}
        />
      </MemoryRouter>
    ));
  }
```

## 0303

- [x]  params 예외처리 추가
- [x]  "/restaurant" ⇒ RestaurantsPage 로 수정
- [x]  restaurantId ⇒ Id 로 수정
- [x]  App -> Page -> Container -> Presentation 로 수정 ( 식당 상세정보 )
- [x]  메뉴와 리스트 출력할 List 컴포넌트 생성
- [x]  의미단위로 enter 쳐서 가독성을 높일 것
- [x]  destructive assin 으로 주소, 정보 출력 변경, test 도 변경
- [x]  RestaurantPage.jsx useEffect 수정
- [x]  restaurantDetail → restaurant 으로 수정

## 0304 

- [x] RestaurantDetailContainer 의 업무를 분산하자
- [x] 각 파일을 해당 폴더로 이동시키기

gyeongwon1275 committed:
:package: Package: package 명령어 추가

gyeongwon1275 committed:
:package: Package: react-router-dom 추가

gyeongwon1275 committed:
:recycle: Refactor: RestaurantPage 생성

- RegionsContainer, CategoriesContainer, RestaurantsContainer 통합 컴포넌트

gyeongwon1275 committed:
:wrench: Config:jest

- verbose true

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- with path /restaurants renders restaurants page

gyeongwon1275 committed:
:construction: Green-Working: react-router App

- react router 적용
- with path /restaurants renders RestaurantPage

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- renderApp
- 헤더링크 여부 확인

gyeongwon1275 committed:
:construction: Green-Working: App

- 헤더 Link 추가

gyeongwon1275 committed:
:white_check_mark: Red-Test: HomePage

- Home heading, About link, Restaurant link 확인

gyeongwon1275 committed:
:construction: Green-Working: HomePage

- add Home

gyeongwon1275 committed:
:white_check_mark: Red-Test: About

- About h2
- About 페이지 입니다 p

gyeongwon1275 committed:
:construction: Green-Working: About

- About heading
- About 페이지 입니다 paragraph

gyeongwon1275 committed:
:white_check_mark: Red-Test: NotFound

- 404 Not Found text 확인

gyeongwon1275 committed:
:construction: Green-Working: NotFound

- h2 404 Not Found

gyeongwon1275 committed:
:wrench: Config: webpack

- devServer historyApiFallback

gyeongwon1275 committed:
:recycle: Refactor: HomePage

- link 를 list 안에 넣음

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- with path / renders HomePage

gyeongwon1275 committed:
:construction: Green-Working: App

- with path / renders HomePage

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- with path /about renders About

gyeongwon1275 committed:
:construction: Green-Working: App

- with path /about renders About

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- with path /any_not_exist_url renders NotFound

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- with path /any_not_exist_url renders NotFound

Kihyun92 commented on src/App.test.jsx:
```diff
@@ -2,33 +2,77 @@ import React from 'react';
+        <App />
+      </MemoryRouter>
+    ));
+  }
+  beforeEach(() => {
```
사이에 공백 추가해주세요~

gyeongwon1275 replied:
수정했습니다~!

Kihyun92 commented on src/App.test.jsx:
```diff
@@ -2,33 +2,77 @@ import React from 'react';
+        { id: 1, name: '마법사주방' },
+      ],
+    }));
+  });
+  context('with path /', () => {
```
사이에 공백 추가해주세요~

gyeongwon1275 replied:
수정했습니다~!

Kihyun92 commented on src/App.test.jsx:
```diff
@@ -2,33 +2,77 @@ import React from 'react';
+      expect(getByRole('link', { name: 'About' })).toHaveAttribute('href', '/about');
+      expect(getByRole('link', { name: 'Restaurants' })).toHaveAttribute('href', '/restaurants');
+    });
+  });
+  context('with path /about', () => {
```
사이에 공백 추가해주세요~

gyeongwon1275 replied:
수정했습니다~!

Kihyun92 reviewed (changes requested):
올바르지 않은 id 값인 경우 어떻게 처리할지에 대한 고민을 해보시는게 좋네요! 👍🏼 

여러 방법들이 있을 것 같아요. 음 일단은 올바르지 않은 요청시 해당 API에서 어떤 값을 내려주는지를 보고 처리할 수도 있고, 또는 프론트에서 API 호출 전에 확인해서 처리를 할지 등이죠 ㅎㅎ

한번 고민해보시면 좋을 부분 같습니다~

gyeongwon1275 committed:
:white_check_mark: Test: Refactor App.test.jsx

- 테스트 함수간 공백 처리

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetail

- 서버에서 데이터를 받아왔을 때 출력하는 지 확인

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetail

- 화면출력

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetailContainer

- selector 함수 추가
- 컴포넌트 감싸는 div 제거

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetail

- 가독성 증가

gyeongwon1275 committed:
:white_check_mark: Red-Test: api

- fetchRestaurantDetail

gyeongwon1275 committed:
:construction: Green-Working: api

- fetchRestaurantDetail

gyeongwon1275 committed:
:white_check_mark: Red-Test: actions

- loadRestaurantDetail
- setRestaurantDetail

gyeongwon1275 committed:
:construction: Green-Working: actions

- loadRestaurantDetail, setRestaurantDetail

gyeongwon1275 committed:
:recycle: Refactor: restaurantDetail mock data 이동

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetailContainer

- url 에서 params 를 받아오는지 확인

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetailContainer

- params 를 받아 mount 시 데이터 요청

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetailContainer

- dispatch(loadRestaurantDetail(restaurantId)) 로 변경

gyeongwon1275 committed:
:package: Package: given2, history 설치

gyeongwon1275 committed:
:white_check_mark: Red-Test: reducer

- setRestaurantDetail
action.type 에 setRestaurantDetail 추가
초기상태에 restaurantDetail 추가

gyeongwon1275 committed:
:construction: Green-Working: reducer

- setRestaurantDetail 추가
- 초기상태에 restaurantDetail 추가

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetail

- reducer 초기상태를 restaurant -> restaurantDetail 로 변경
이유 : restaurants 와 확실히 구분하기 위함

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetailContainer, RestaurantDetail

- 초기상태 key 명 변경으로 인한 수정
useSelector , prop 명 변경

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetailContainer

- hooks 간 간격 추가

gyeongwon1275 committed:
:white_check_mark: Red-Test:RestaurantsContainer

- 음식점을 link 로 변경

gyeongwon1275 committed:
:construction: Green-Working:RestaurantsContainer

- 음식점을 link로 변경

gyeongwon1275 committed:
:white_check_mark: Test: App

- link 안에 있는 text 를 찾아서 href 올바른지 확인

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- with path /restaurants/:restaurantId renders restaurant detail page

gyeongwon1275 committed:
:construction: Green-Working: App

- with path /restaurants/:restaurantId renders restaurant detail page

gyeongwon1275 committed:
:white_check_mark: Test: Refactor RestaurantDetail

- 기존 별도로 분리된 dispatch 여부 test 를 하나로 합침

gyeongwon1275 committed:
:white_check_mark: Red-Test: App

- url /restaurant/4 -> /restaurants/4 수정

gyeongwon1275 committed:
:construction: Green-Working: APP

- /restaurants/:restaurantId 로 url 수정

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantsContainer

- /1 => /restaurants/1 으로 수정

gyeongwon1275 committed:
:construction: Green-Working: RestaurantsContainer

- /restaurant.id => /restaurants/${restaurant.id} 으로 수정

gyeongwon1275 committed:
:white_check_mark: Red-Test: reducer

- reducer restaurantDetail 초기상태를 null 로 지정

gyeongwon1275 committed:
:construction: Green-Working: reducer

- restaurantDetail 초기값을 null 로 지정

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetailContainer

- restaurantDetail 없을 때 테스트 추가
- given2 적용

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetail

- restaurantDetail 없으면 "텅~!" 출력

gyeongwon1275 commented:
# 질문

## 1> 

**path="/restaurants/:restaurantId" 일 때 params 를 받지 못하고 앱이 멈추는 현상 발생**

react-router link 를 클릭해서 /restaurant/:restaurantId 형식으로 접근하면 
restaurantId 받은 후 화면에 출력하는데 

직접 주소창에 /restaurant/:restaurantId 을 입력하면 아래와 같이 나오며 앱이 멈춰버립니다.. 
GET http://localhost:8080/restaurants/main.js net::ERR_ABORTED 404 (Not Found)

일단 확인해 본 것은 /:restaurantId 형식으로 직접 url 에 입력하면 restaurantId 을 받을 수 있다는 것을 확인하였습니다.

공식문서 (https://reactrouter.com/web/example/url-params) 에서 실험한 결과 제대로 동작하는 것을 보면  
예상되는 문제는 webpack 설정 문제인 것 같은데 제가 뭘 잘못하고 있는 걸까요?? 

## 2> 

**현재 url 에서 params 을 받아오는지를 테스트 해야하는지** 

아래처럼 history 객체를 넘겨주어 테스트해야하는지 아니면 useParams 를 mock 처리할지 궁금합니다. 

```javascript
 function renderRestaurantDetail(path) {
    const history = createMemoryHistory();
    return render((
      <MemoryRouter initialEntries={[path]}>
        <Route
          exact
          history={history}
          path="/restaurants/:restaurantId"
          component={RestaurantDetailContainer}
        />
      </MemoryRouter>
    ));
  }
```

gyeongwon1275 committed:
:bug: Fix: index.html

Kihyun92 commented on src/App.jsx:
```diff
@@ -1,32 +1,27 @@
+      <Switch>
+        <Route exact path="/" component={HomePage} />
+        <Route exact path="/about" component={About} />
+        <Route exact path="/restaurants" component={RestaurantPage} />
```
`RestaurantsPage` 가 맞을 것 같아요~

gyeongwon1275 replied:
RestaurantsPage 로 수정했습니다~!

Kihyun92 commented on src/App.jsx:
```diff
@@ -1,32 +1,27 @@
+        <Route exact path="/" component={HomePage} />
+        <Route exact path="/about" component={About} />
+        <Route exact path="/restaurants" component={RestaurantPage} />
+        <Route exact path="/restaurants/:restaurantId" component={RestaurantDetailContainer} />
```
`path="/restaurants/:id"` 이렇게만 처리해주는건 어떨까요?

gyeongwon1275 replied:
앞에 /restaurants 가 있기 때문에 굳이 restaurantId 가 필요 없겠네요 id 로 수정했습니다 ~!

Kihyun92 commented on src/RestaurantDetailContainer.jsx:
```diff
@@ -0,0 +1,25 @@
+import RestaurantDetail from './RestaurantDetail';
+import { loadRestaurantDetail } from './actions';
+
+export default function RestaurantDetailContainer() {
```
`container-presentation`을 잘 나눠주신 것 같아요~

라우팅에 대한 부분도 따로 관리해주는 건 어떤가요?

이런 순서일 것 같아요. `App -> Page -> Container -> Presentation`

gyeongwon1275 replied:
Page -> 라우터 담당 , Container-> api 등 데이터 담당, Presentation -> view 담당으로 지정해서 수정했습니다~!

Kihyun92 commented on src/RestaurantDetail.jsx:
```diff
@@ -0,0 +1,41 @@
+export default function RestaurantDetail({ restaurantDetail }) {
+  if (!restaurantDetail) {
+    return <h3>텅~!</h3>;
+  }
+  const {
```
라인 사이에 공백을 추가해주는게 좋을 것 같아요~

gyeongwon1275 replied:
공백추가 하는 걸로 수정했습니다!

Kihyun92 commented on src/RestaurantDetail.jsx:
```diff
@@ -0,0 +1,41 @@
+  }
+  const {
+    name, address, menuItems, reviews, information,
+  } = restaurantDetail;
+  return (
```
라인 사이에 공백을 추가해주는게 좋을 것 같아요~

gyeongwon1275 replied:
공백추가 하는 걸로 수정했습니다!

Kihyun92 commented on src/RestaurantDetail.jsx:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <div>
+      <h2>{name}</h2>
+      <div>
+        {'주소 : '}
+        <p>{address}</p>
+      </div>
```
```suggestion
      <div>
        {`주소 : ${address}`}
      </div>
```

`template literal`로 처리해주는건 어떤가요? 의도와 다를까요?
- https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Template_literals

gyeongwon1275 replied:
address 만 test 하려고 하다 보니 구조분해 할당하면 getByText 안걸려서 따로 태그를 분리했었는데 
getByText(`주소 : ${address}`) 로 test 수정하고 template literal 적용했습니다!

Kihyun92 commented on src/RestaurantDetail.jsx:
```diff
@@ -0,0 +1,41 @@
+
+      <h3>메뉴</h3>
+      <ul>
+        {menuItems.map((menu) => (
```
구조 분해 할당으로 처리해줘도 좋겠네요~

gyeongwon1275 replied:
```javascript
 const {
    name, address, menuItems, reviews, information,
  } = restaurant;
```
구조분해 할당하려면 위에 name 과 변수가 겹처서 메뉴, 리뷰 리스트 부분을 따로 컴포넌트로 분리했습니다.

Kihyun92 commented on src/RestaurantPage.jsx:
```diff
@@ -0,0 +1,25 @@
+export default function RestaurantPage() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(loadInitialData());
+  });
```
`useEffect`에 어떤 값들이 변할 때 호출될건지 써줘야할 것 같아요. 한 번만 실행하고 싶다면 빈 배열을 넣어주면 됩니다~

gyeongwon1275 replied:
처음 mount 될 때 한번만 하면 될 것같아 빈배열로 수정했습니다.

Kihyun92 commented on src/RestaurantsContainer.jsx:
```diff
@@ -11,7 +13,9 @@ export default function RestaurantsContainer() {
     <ul>
       {restaurants.map((restaurant) => (
```
구조 분해 할당으로 처리해줘도 좋겠군요~

gyeongwon1275 replied:
destructuring assign 안되있는 다른 부분까지 수정했습니다!

Kihyun92 commented on src/reducer.js:
```diff
@@ -6,6 +6,7 @@ const initialState = {
   restaurants: [],
   selectedRegion: null,
   selectedCategory: null,
+  restaurantDetail: null,
```
레스토랑 정보에 해당하는 상태인 것 같아요~ `restaurant`라고만 하는건 어떤가요?

사용하는 곳에서 보면 `restaurantDetail.address`에서 `restaurant.address`로 될 것 같네요.

gyeongwon1275 replied:
restaurants 와 헷갈릴 것 같아 restaurantDetail 로 설정했었는데 restaurant 으로 수정했습니다!

Kihyun92 reviewed (changes requested):
> 현재 url 에서 params 을 받아오는지를 테스트 해야하는지 
아래처럼 history 객체를 넘겨주어 테스트해야하는지 아니면 useParams 를 mock 처리할지 궁금합니다.

공식 문서를 잘 확인해보신 것 같아요 👍🏼  

말씀하신 것과 같은 몇가지 방법이 있을 것 같은데, 정답은 없는 부분인 것 같아요. 

일단 공식 문서대로 한번 해보시고 잘 해결 된다면 슬랙 채널에 공유해도 좋겠네요 :)

gyeongwon1275 committed:
:recycle: Refactor: App, RestaurantPage

- RestaurantPage -> RestaurantsPage 로 수정

gyeongwon1275 committed:
:recycle: Refactor: restaurantId -> id 로 params 수정

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetail

- 가독성을 높이기 위한 enter 처리

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetailContainer

- 주소, 정보 destructive assign 적용

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetail

- 주소, 정보 구조분해할당

gyeongwon1275 committed:
:white_check_mark: Test: App

- 주소, 정보 destructive assign 적용

gyeongwon1275 committed:
:recycle: Refactor: restaurantDetail -> restaurant 로 변경

- api, action reducer prop 등 연관된 부분전체 변경

gyeongwon1275 committed:
:white_check_mark: Red-Test: List

- 메뉴, 리뷰 출력용list 컴포넌트

gyeongwon1275 committed:
:construction: Green-Working: List

- 메뉴, 리뷰 출력용 컴포넌트 생성

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetail

- List component 적용

gyeongwon1275 committed:
:recycle: Refactor: App

- router 담당할 RestaurantDetailPage 적용

gyeongwon1275 committed:
:recycle: Refactor:RegionsContainer

- view 담당 컴포넌트인 Regions 를 적용

gyeongwon1275 committed:
:recycle: Refactor: CategoriesContainer

- view 만담당하는 Categories 적용

gyeongwon1275 committed:
:hammer: Update: RestaurantsPage

- mount 될 때 한번만 초기 지역, 카테고리 목록 불러오기

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetailContainer

- 잘못된 params 입력시 예외처리

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetailContainer

- 잘못된 params 입력시 예외처리

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetailContainer

- 조건을 확실하게 수정

gyeongwon1275 committed:
:recycle: Refactor: 가독성 증가

- 가독성위한 줄바꿈

gyeongwon1275 committed:
:recycle: Refactor: Categories

- id,name => destructuring assign

gyeongwon1275 committed:
:recycle: Refactor: Regions

- estructuring assign id name

gyeongwon1275 committed:
:recycle: Refactor: RestaurantsContainer

- destructuring assign id,name

gyeongwon1275 committed:
:recycle: Refactor: RestaurantsContainer

- view 를 별도 컴포넌트로 분리

gyeongwon1275 committed:
:recycle: Refactor: RestaurantDetailContainer

- 부정조건문을 사용하지 않도록 수정

Kihyun92 commented on src/RestaurantDetailPage.jsx:
```diff
@@ -0,0 +1,9 @@
+
+import RestaurantDetailContainer from './RestaurantDetailContainer';
+
+export default function RestaurantDetailPage() {
```
이 페이지 컴포넌트의 역할이 무엇인가요? 단순히 컴포넌트만 연결해주고 있는 것 같군요.

`RestaurantDetailContainer` 의 역할을 나눌 수 있을까요?

gyeongwon1275 replied:
>이 페이지 컴포넌트의 역할이 무엇인가요? 단순히 컴포넌트만 연결해주고 있는 것 같군요.

지적하신 것 처럼 RestaurantDetailPage 가 아무역할도 하지 않는 반면 RestaurantDetailContainer 에 과도한 역할이 몰려있다고 생각했습니다. 

이에 RestaurantDetailContainer 는 store state 에서 데이터를 받아서 전달하는 역할 

RestaurantDetailPage 는 useParams 를 받아서 처리하는 역할로 구분했습니다.

Kihyun92 reviewed (changes requested):
src 폴더 내에 파일들이 꽤 많아진 것 같아요~ 의도와 역할에 따라 폴더로 구분해보는건 어떨까요?

잘 진행해주고 계셔서 어떤걸 리뷰를 더 드려야할지 고민되는군요 😂

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetailContainer

- RestaurantDetailContainer 에서는 store 상태에서 식당정보를 받아오는 것만을 수행

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetailContainer

- store 에서 식당정보를 가져와서 전달하는 역할만 수행하는 것으로 수정

gyeongwon1275 committed:
:white_check_mark: Red-Test: RestaurantDetailPage

- params 받아서 NotFound 출력할지 RestaurantDetailContainer 출력할지
결정하는 역할을 수행하도록 함

gyeongwon1275 committed:
:construction: Green-Working: RestaurantDetailPage

- params 받아서 어떤 컴포넌트를 출력할지 결정하는 역할
- params 를 서버에 요청하는 역할

gyeongwon1275 commented:
>이 페이지 컴포넌트의 역할이 무엇인가요? 단순히 컴포넌트만 연결해주고 있는 것 같군요.

지적하신 것 처럼 RestaurantDetailPage 가 아무역할도 하지 않는 반면 RestaurantDetailContainer 에 과도한 역할이 몰려있다고 생각했습니다. 

이에 RestaurantDetailContainer 는 store state 에서 데이터를 받아서 전달하는 역할 

RestaurantDetailPage 는 Route 에 컴포넌트로 지정했기 때문에 useParams 를 받아서 처리하는 역할로 구분했습니다. 

올바르게 구분한 걸까요?

gyeongwon1275 committed:
:truck: Resource: page,components,container 로 분류하여 폴더이동

gyeongwon1275 committed:
:truck: Resource: 폴더명 뒤에 s 붙임

Kihyun92 commented on src/containers/RegionsContainer.jsx:
```diff
@@ -0,0 +1,34 @@
+export default function RegionsContainer() {
+  const dispatch = useDispatch();
+
+  const regions = useSelector(get('regions'));
+
+  const selectedRegion = useSelector(get('selectedRegion'));
```
사이에 공백은 빼줘도 좋을 것 같네요~

Kihyun92 commented on src/containers/RegionsContainer.jsx:
```diff
@@ -0,0 +1,34 @@
+      selectedRegion={selectedRegion}
+      onClick={handleClick}
+    />
+
```
이곳에 공백은 제거해도 될 것 같네요 ㅎㅎ

Kihyun92 commented on src/containers/RestaurantsContainer.jsx:
```diff
@@ -1,19 +1,14 @@
 import React from 'react';
 
 import { useSelector } from 'react-redux';
+import Restaurants from '../components/Restaurants';
```
사이에 공백을 추가해서 외부 모듈과 컴포넌트 간에 구분을 해주는것도 좋을 것 같네요

Kihyun92 commented on src/pages/RestaurantDetailPage.jsx:
```diff
@@ -0,0 +1,30 @@
+
+  const { id } = useParams();
+
+  const isInValidParam = !Number.isInteger(parseInt(id, 10));
```
컴포넌트 바깥에 함수로 빼도 좋겠네요 :)

Kihyun92 reviewed (changes requested):
### 1번 답변
넵, 좋은 것 같습니다! 👍🏼 

### 2번 답변
이 부분은 상황마다 다를 것 같은데, 지금의 구조에서는 __tests__ 폴더로 분리하지 않아도 괜찮을 것 같네요(지금 구조도 괜찮습니다 ㅎㅎ)

그 이유는 컴포넌트가 다른 폴더로 옮겨지는 경우, 테스트 파일에서 경로 수정해주는 걸 놓칠 수 있을것 같아요.

만약 컴포넌트 별로 폴더가 따로 관리된다면 (Restaurant 폴더 안에 Restaurant.jsx, css 파일 등이 있는 구조) 그 안에서 테스트 폴더를 만들고 테스트를 관리해도 상관없겠죠. 이 경우엔 컴포넌트를 옮기면 같이 옮겨질 테니까요 ㅎㅎ

마지막 리뷰가 될 것 같네요. 혹시 리뷰 받고 싶은 부분이 있다면 언제든 편하게 말씀해주세요 :)

Kihyun92 commented:
이번 주도 고생하셨습니다~ 

위에 커맨트 남긴 부분 한번 확인해보시면 좋을 것 같고, 오늘 밤~내일 오전 중에 올라올 과제 풀이 영상 꼭 보셨으면 좋겠네요 :)
